x-pack/platform/plugins/shared/integration_assistant/public/components/create_integration/create_integration_assistant/flyout/cel_configuration/steps/confirm_settings_step/confirm_settings_step.tsx

```diff
@@ -0,0 +1,365 @@
+  const authMethods = apiSpec?.operation(path, 'get').prepareSecurity();
+  const specifiedAuth = authMethods ? Object.keys(authMethods) : [];
+  return specifiedAuth;
+};
+
+interface ConfirmSettingsStepProps {
+  integrationSettings: State['integrationSettings'];
+  connector: State['connector'];
+  isFlyoutGenerating: State['isFlyoutGenerating'];
+  suggestedPaths: string[];
+  showValidation: boolean;
+  onShowValidation: () => void;
+  onUpdateValidation: (updatedIsValid: boolean) => void;
+  onUpdateNeedsGeneration: (updatedNeedsGeneration: boolean) => void;
+  onCelInputGenerationComplete: (path: string, auth: CelAuthType, celInputResult: CelInput) => void;
+}
+
+export const ConfirmSettingsStep = React.memo<ConfirmSettingsStepProps>(
+  ({
+    integrationSettings,
+    connector,
+    isFlyoutGenerating,
+    suggestedPaths,
+    showValidation,
+    onShowValidation,
+    onUpdateValidation,
+    onUpdateNeedsGeneration,
+    onCelInputGenerationComplete,
+  }) => {
+    const { setIsFlyoutGenerating } = useActions();
+    const { http, notifications } = useKibana().services;
+    const { reportCelGenerationComplete } = useTelemetry();
+
+    const [selectedPath, setSelectedPath] = useState<string>();
+    const [selectedOtherPath, setSelectedOtherPath] = useState<string | undefined>();
+    const [useOtherPath, setUseOtherPath] = useState<boolean>(false);
+    const coalescedSelectedPath = !useOtherPath ? selectedPath : selectedOtherPath;
+
+    const [selectedAuth, setSelectedAuth] = useState<string | undefined>();
+
+    const [specifiedAuthForPath, setSpecifiedAuthForPath] = useState<string[]>([]);
+    const [unspecifiedAuth, setUnspecifiedAuth] = useState<boolean>(false);
+
+    const [successfulGeneration, setSuccessfulGeneration] = useState<boolean>(false);
+    const [generatedPair, setGeneratedPair] = useState<{
+      path: string | undefined;
+      auth: CelAuthType | undefined;
+    }>({ path: undefined, auth: undefined });
+    const [error, setError] = useState<null | string>(null);
+
+    const [fieldValidationErrors, setFieldValidationErrors] = useState({
+      path: false,
+      auth: false,
+    });
+
+    const isSelectedPathGenerated =
+      generatedPair.path === coalescedSelectedPath &&
+      generatedPair.auth ===
+        (selectedAuth && translateDisplayAuthToType(selectedAuth).toLowerCase());
+
+    // sets the recommended options on load
+    useEffect(() => {
+      if (!selectedPath) {
+        const recommendedPath = suggestedPaths ? suggestedPaths[0] : '';
+        setSelectedPath(recommendedPath);
+        if (recommendedPath) {
+          const specifiedAuth = getSpecifiedAuthForPath(
+            integrationSettings?.apiSpec,
+            recommendedPath
+          );
+          setSelectedAuth(translateAuthTypeToDisplay(specifiedAuth[0]));
+        }
+      }
+    }, [integrationSettings?.apiSpec, selectedPath, suggestedPaths]);
```

Comment:
If this is needed only to set a value on load, it is better to use the useState parameter:

```ts
const [selectedPath, setSelectedPath] = useState<string>(() => suggestedPaths ? suggestedPaths[0] : '');
```

```ts
const [selectedAuth, setSelectedAuth] = useState<string | undefined>(() => {
  const recommendedPath = suggestedPaths ? suggestedPaths[0] : '';
  if (recommendedPath) {
    const specifiedAuth = getSpecifiedAuthForPath(
      integrationSettings?.apiSpec,
      recommendedPath
    );
    return translateAuthTypeToDisplay(specifiedAuth[0]);
  }
});
```

So they are executed only once. However, if `suggestedPaths` is updated this logic won't be re-executed, unless the component is entirely re-rendered.

Reply:
Ooh thanks for sharing! Will update